### PR TITLE
Let with multiple bindings

### PR DESCRIPTION
**Pull request by kroening**
- [X] Each commit message has a non-empty body, explaining why the change was made.
- n/a Methods or procedures I have added are documented, following the guidelines provided in CODING_STANDARD.md.
- n/a The feature or user visible behaviour I have added or modified has been documented in the User Guide in doc/cprover-manual/
- [ ] Regression or unit tests are included, or existing tests cover the modified code (in this case I have detailed which ones those are in the commit message).
- n/a My commit message includes data points confirming performance improvements (if claimed).
- [ ] My PR is restricted to a single feature or bugfix.
- n/a White-space or formatting changes outside the feature-related changed lines are in commits of their own.

### CODE REVIEW

**Review by allredj — COMMENTED** [review]
:heavy_check_mark:
Passed Diffblue compatibility checks (cbmc commit: a76f57ba529bf1a51661667a8476b007f616b44f).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/131490735

File: src/solvers/prop/prop_conv_solver.cpp [hunk 1]
```diff
@@ -308,23 +310,36 @@ literalt prop_conv_solvert::convert_bool(const exprt &expr)
...
 
     // first check whether this is all boolean
-    if(
-      let_expr.value().type().id() == ID_bool &&
-      let_expr.where().type().id() == ID_bool)
+    bool all_boolean = let_expr.where().type().id() == ID_bool;
+
+    for(auto &v : values)
+      if(v.type().id() != ID_bool)
+        all_boolean = false;
```
Comment (romainbrenguier) [inline src/solvers/prop/prop_conv_solver.cpp]: could use `std::all_of`

File: src/solvers/prop/prop_conv_solver.cpp [hunk 2]
```diff
@@ -308,23 +310,36 @@ literalt prop_conv_solvert::convert_bool(const exprt &expr)
...
+
+    if(all_boolean)
     {
-      literalt value = convert(let_expr.value());
+      // yes, it's all boolean
```
Comment (romainbrenguier) [inline src/solvers/prop/prop_conv_solver.cpp]: comment unnecessary (the `if(all_boolean)` is clear enough)

**Review by romainbrenguier — APPROVED** [review]

**Review by allredj — COMMENTED** [review]
:heavy_check_mark:
Passed Diffblue compatibility checks (cbmc commit: cd0d0f6c95c298305d696fbbff82934dac43719f).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/131602265

File: src/solvers/prop/prop_conv_solver.cpp (hunk 1 above)
Comment (kroening) [reply src/solvers/prop/prop_conv_solver.cpp]: sure

**Review by peterschrammel — APPROVED** [review]

**Pull request merged**